src/workload.py

```diff
@@ -0,0 +1,238 @@
+        if not self._enabled:
+            return
+        logger.debug("Disabling MySQL Router service")
+        self.shell.delete_user(self._router_username)
+        self._update_layer(enabled=False)
+        logger.debug("Disabled MySQL Router service")
+
+    def _restart(self, *, tls: bool) -> None:
+        """Restart MySQL Router to enable or disable TLS."""
+        logger.debug("Restarting MySQL Router service")
+        router_password = self.shell.change_mysql_router_user_password(self._router_username)
+        self._bootstrap_router(password=router_password, tls=tls)
+        logger.debug("Restarted MySQL Router service")
+        self._charm.wait_until_mysql_router_ready()
+
+    def _write_file(self, path: pathlib.Path, content: str) -> None:
+        """Write content to file.
+
+        Args:
+            path: Full filesystem path (with filename)
+            content: File content
+        """
+        self._container.push(
+            str(path),
+            content,
+            permissions=0o600,
+            user=self._UNIX_USERNAME,
+            group=self._UNIX_USERNAME,
+        )
+        logger.debug(f"Wrote file {path=}")
+
+    def _delete_file(self, path: pathlib.Path) -> None:
+        """Delete file.
+
+        Args:
+            path: Full filesystem path (with filename)
+        """
+        path = str(path)
+        if self._container.exists(path):
+            self._container.remove_path(path)
+            logger.debug(f"Deleted file {path=}")
+
+    @property
+    def _tls_config_file(self) -> str:
+        """Render config file template to string.
+
+        Config file enables TLS on MySQL Router.
+        """
+        with open("templates/tls.cnf", "r") as template_file:
+            template = string.Template(template_file.read())
+        config_string = template.substitute(
+            tls_ssl_key_file=self._ROUTER_CONFIG_DIRECTORY / self._TLS_KEY_FILE,
+            tls_ssl_cert_file=self._ROUTER_CONFIG_DIRECTORY / self._TLS_CERTIFICATE_FILE,
+        )
+        return config_string
+
+    def enable_tls(self, *, key: str, certificate: str):
+        """Enable TLS and restart MySQL Router service."""
+        logger.debug("Enabling TLS")
+        self._write_file(
+            self._ROUTER_CONFIG_DIRECTORY / self._TLS_CONFIG_FILE, self._tls_config_file
+        )
+        self._write_file(self._ROUTER_CONFIG_DIRECTORY / self._TLS_KEY_FILE, key)
+        self._write_file(self._ROUTER_CONFIG_DIRECTORY / self._TLS_CERTIFICATE_FILE, certificate)
+        if self._enabled:
+            self._restart(tls=True)
+        logger.debug("Enabled TLS")
+
+    def disable_tls(self) -> None:
+        """Disable TLS and restart MySQL Router service."""
+        logger.debug("Disabling TLS")
+        for file in [self._TLS_CONFIG_FILE, self._TLS_KEY_FILE, self._TLS_CERTIFICATE_FILE]:
+            self._delete_file(self._ROUTER_CONFIG_DIRECTORY / file)
+        if self._enabled:
+            self._restart(tls=False)
```

Comment:
do we need exception handling as `wait_until_mysql_router_ready` can raise an assertion error?
same question for enable_tls